pkg/clusteragent/admission/controllers/webhook/controller.go

```diff
@@ -0,0 +1,260 @@
+// It can be a callback function for update events.
+func (c *Controller) handleSecretUpdate(oldObj, newObj interface{}) {
+	if !c.isLeaderFunc() {
+		return
+	}
+	newSecret, ok := newObj.(*corev1.Secret)
+	if !ok {
+		log.Debugf("Expected Secret object, got: %v", newObj)
+		return
+	}
+	oldSecret, ok := oldObj.(*corev1.Secret)
+	if !ok {
+		log.Debugf("Expected Secret object, got: %v", oldObj)
+		return
+	}
+	if newSecret.ResourceVersion == oldSecret.ResourceVersion {
+		return
+	}
+	c.handleSecret(newObj)
+}
+
+// handleWebhook enqueues the targeted Webhook object when an event occurs.
+// It can be a callback function for deletion and addition events.
+func (c *Controller) handleWebhook(obj interface{}) {
+	if !c.isLeaderFunc() {
+		return
+	}
+	if object, ok := obj.(metav1.Object); ok {
+		if object.GetName() == c.config.GetName() {
+			c.enqueue(object)
+		}
+	}
+}
+
+// handleWebhookUpdate handles the new Webhook reported in update events.
+// It can be a callback function for update events.
+func (c *Controller) handleWebhookUpdate(oldObj, newObj interface{}) {
+	if !c.isLeaderFunc() {
+		return
+	}
+	newWebhook, ok := newObj.(*admiv1beta1.MutatingWebhookConfiguration)
+	if !ok {
+		log.Debugf("Expected MutatingWebhookConfiguration object, got: %v", newObj)
+		return
+	}
+	oldWebhook, ok := oldObj.(*admiv1beta1.MutatingWebhookConfiguration)
+	if !ok {
+		log.Debugf("Expected MutatingWebhookConfiguration object, got: %v", oldObj)
+		return
+	}
+	if newWebhook.ResourceVersion == oldWebhook.ResourceVersion {
+		return
+	}
+	c.handleWebhook(newObj)
+}
+
+// enqueue adds a given object to the work queue.
+func (c *Controller) enqueue(obj interface{}) {
+	key, err := cache.MetaNamespaceKeyFunc(obj)
+	if err != nil {
+		log.Debugf("Couldn't get key for object %v: %v, adding it to the queue with an unnamed key", obj, err)
+		c.queue.Add(struct{}{})
+		return
+	}
+	log.Debugf("Adding object with key %s to the queue", key)
+	c.queue.Add(key)
+}
+
+// requeue adds an object's key to the work queue for
+// a retry if the rate limiter allows it.
+func (c *Controller) requeue(key interface{}) {
+	c.queue.AddRateLimited(key)
+}
+
+// run waits for items to process in the work queue.
+func (c *Controller) run() {
+	for c.processNextWorkItem() {
+	}
+}
+
+// processNextWorkItem handle the reconciliation
+// of the Webhook when new item is added to the work queue.
+// Always returns true unless the work queue was shutdown.
+func (c *Controller) processNextWorkItem() bool {
+	key, shutdown := c.queue.Get()
+	if shutdown {
+		return false
+	}
+	defer c.queue.Done(key)
+
+	if err := c.reconcile(); err != nil {
+		c.requeue(key)
+		log.Infof("Couldn't reconcile Webhook %s: %v", c.config.GetName(), err)
+		return true
+	}
+
+	c.queue.Forget(key)
+	log.Debugf("Webhook %s reconciled successfully", c.config.GetName())
+
+	return true
+}
+
+// reconcile reconciles the current state of the Webhook with its desired state.
+func (c *Controller) reconcile() error {
```

Comment:
It might be a bit too much to roll out a full blown controller to handle 1 secret/webhook. Not even sure keeping an Informer in memory is worth it. A simple loop every X sound more than enough no?
I think it's the same for the secret controller.